docs/customizing.rst

```diff
@@ -6,75 +6,101 @@ Customizing
 ===========
 
 Flask-SQLAlchemy defines sensible defaults. However, sometimes customization is
-needed. Two major pieces to customize are the Model base class and the default
-Query class.
+needed. There are various ways to customize how the models are defined and
+interacted with.
 
-Both of these customizations are applied at the creation of the :class:`SQLAlchemy`
+These customizations are applied at the creation of the :class:`SQLAlchemy`
 object and extend to all models derived from its ``Model`` class.
 
+
 Model Class
 -----------
 
-Flask-SQLAlchemy allows defining a custom declarative base, just like SQLAlchemy,
-that all model classes should extend from. For example, if all models should have
-a custom ``__repr__`` method::
+SQLAlchemy models all inherit from a declarative base class. This is exposed
+as ``db.Model`` in Flask-SQLAlchemy, which all models extend. This can be
+customized by subclassing the default and passing the custom class to
+``model_class``.
 
-    from flask_sqlalchemy import Model  # this is the default declarative base
-    from flask_sqlalchemy import SQLAlchemy
+For example, to allow testing model instances for equality based on their
+primary keys::
+
+    from flask_sqlalchemy import Model, SQLAlchemy
+    import sqlalchemy as sa
 
-    class ReprBase(Model):
-         def __repr__(self):
-             return "<{0} id: {1}>".format(self.__class__.__name__, self.id)
+    class EqModel(Model):
+        def __eq__(self, other):
+            if not isinstance(other, type(self)):
+                return NotImplemented
 
-    db = SQLAlchemy(model_class=ReprBase)
+            return sa.inspect(self).identity == sa.inspect(other).identity
 
-    class MyModel(db.Model):
+    db = SQLAlchemy(model_class=EqModel)
+
+    class Language(db.Model):
         ...
 
-.. note::
+    Language.query.filter_by(name='Python') == Language.query.get(1)
```

Comment:
especially for this example there is no need for a custom `__eq__` (actually, the two objects are event identical, not just equal)